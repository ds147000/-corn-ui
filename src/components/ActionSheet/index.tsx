/* eslint-disable import/export */

import ActionSheet  from './actionSheet'

export { ActionSheetItem } from './item'
export default ActionSheet
