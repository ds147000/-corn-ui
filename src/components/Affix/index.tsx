/* eslint-disable no-unreachable */
/* eslint-disable no-magic-numbers */
// #if _APP === 'weapp'
import Taro, { usePageScroll } from '@tarojs/taro'
// #endif
import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
import { View } from '@tarojs/components'
import ClassNames from 'classnames'
import { ViewProps } from '../../types/View'
import { judge } from './utils/judge'

export interface AffixProps extends ViewProps {
  position?: 'top' | 'bottom' | 'right' | 'left'
  onChange?(fixed: boolean): void
}

const Affix: React.FC<AffixProps> = ({ children, position = 'top', onChange, className, ...props }) => {
  const [ fixed, setFixed ] = useState(false)
  const [ boxStyle, setBoxStyle ] = useState<React.CSSProperties>({})
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const ref = useRef<any>()

  const oldFixed = useRef<boolean>(false)

  const _class = useMemo(() => {
    return ClassNames('corn-affix', fixed && `corn-affix-${position}`, className)
  }, [ fixed, position, className ])

  const changeFixed = useCallback((value: boolean): void => {
    setFixed((state) => {
      oldFixed.current = state
      return value
    })
  }, [])

  // #if _APP === 'weapp'
  const system = useRef(Taro.getSystemInfoSync())
  const timer = useRef<NodeJS.Timeout>()

  const checkFlexd = (): void => {
    const node = ref.current

    const rootRect = {
      top: 0,
      left: 0,
      bottom: system.current.windowHeight,
      right: system.current.windowWidth
    } as DOMRectReadOnly

    Taro.createSelectorQuery()
      .select('#' + node.uid)
      .fields({ size: true, rect: true }, (res) => {
        if (!res) return

        setBoxStyle({ height: res?.height, width: res?.width })
        const result = judge(res as unknown as DOMRect, rootRect, position, 0)
        changeFixed(result)
      })
      .exec()
  }

  usePageScroll(() => {
    if (timer.current) clearTimeout(timer.current)
    timer.current = setTimeout(() => checkFlexd(), 10)
  })

  useEffect(() => {
    return (): void => {
      if (timer.current) clearTimeout(timer.current)
    }
  }, [])

  // #else
  useEffect(() => {
    // eslint-disable-next-line no-unreachable
    const el: HTMLDivElement = ref.current
    const rootBounds = {
      x: 0,
      y: 0,
      width: window.innerWidth,
      height: window.innerHeight,
      top: 0,
      left: 0,
      bottom: window.innerHeight,
      right: window.innerWidth
    } as DOMRectReadOnly

    setBoxStyle({ height: el.offsetHeight, width: el.offsetWidth })

    function handleScroll(): void {
      const item = el.getBoundingClientRect()
      const result = judge(item, rootBounds, position, 0)
      changeFixed(result)
    }
    handleScroll()
    window.addEventListener('scroll', handleScroll)
    return (): void => {
      window.removeEventListener('scroll', handleScroll)
    }
  }, [ position, changeFixed ])
  // #endif

  useEffect(() => {
    if (oldFixed.current !== fixed) {
      oldFixed.current = fixed
      onChange?.(fixed)
    }
  }, [ fixed, onChange ])

  return (
    <View style={boxStyle} ref={ref}>
      <View className={_class} {...props} >
        {children}
      </View>
    </View>
  )
}

export default Affix
