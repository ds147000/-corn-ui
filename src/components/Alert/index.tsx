import React from 'react'
import { View } from '@tarojs/components'
import classNames from 'classnames'
import { ViewProps } from '../../types/View'
import Icon from '../Icon'

const Alter: React.FC<ViewProps> = ({ children, className, ...props }) => {
  const _class = classNames(className, 'corn-alert', 'corn-f')
  return (
    <View className={_class} {...props} >
      <Icon name="error" className="icon" />
      {children}
    </View>
  )
}

export default Alter
