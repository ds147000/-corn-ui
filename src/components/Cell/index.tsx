import React, { useCallback } from 'react'
import ClassNames from 'classnames'
import { View, Text } from '@tarojs/components'
import { ViewProps } from '@tarojs/components/types/View'
import Icon from '../Icon'
import Link, { LinkProps } from '../Link'
import CellList from './list'

export interface CellProps extends ViewProps, LinkProps {
  /** 标题 */
  label?: string | React.ReactNode
  /** 内容 */
  value?: string | React.ReactNode
  /** 内容为空占位提示 */
  placeholder?: string | React.ReactNode
  /** 是否显示箭头 */
  arrow?: boolean
  /** 尾缀内容 */
  suffix?: React.ReactNode | string
  /** 禁用 */
  disabled?: boolean
  /** 跳转链接，填写启用A链接行为。同时会显示箭头 */
  to?: string
}

type CellView =  { List: typeof CellList }



const Cell: React.FC<CellProps> & CellView = ({
  label, value, arrow, suffix,
  disabled, to, children, onClick,
  className, placeholder, ...props
}) => {

  const _onClick = useCallback((event): void => {
    if (disabled) return
    onClick?.(event)
  }, [ onClick, disabled ])

  const _props: ViewProps = {
    className: ClassNames('xrk-cell xrk-f xrk-ac', disabled && 'xrk-cell-disable', className),
    onClick: _onClick,
    ...props
  }

  const showArrow = (arrow || Boolean(to)) && arrow !== false
  const showPlaceholder = Boolean(value) === false && Boolean(children) === false

  const content = (
    <>
      <View className="xrk-cell-label">{label}</View>
      <View className="xrk-cell-value">
        {value}{children}
        {showPlaceholder && <Text className="xrk-cell-placeholder">{placeholder}</Text>}
      </View>
      <View className="xrk-cell-suffix xrk-f xrk-ac">
        {suffix}
        {showArrow && <Icon name="big-right" data-testid="arrow" />}
      </View>
    </>
  )

  if (to)
    return (<Link {..._props} to={to} disabled={disabled} target="View" >{content}</Link>)

  return (<View {..._props} >{content}</View>)
}

Cell.List = CellList

export default Cell
