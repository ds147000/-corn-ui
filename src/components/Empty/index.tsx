import React from 'react'
import classNames from 'classnames'
import { View, Image, Text } from '@tarojs/components'
import { transformRem } from '../../utils'

const PRESENTED_IMAGE_DEFAULT = 'https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png'
const PRESENTED_IMAGE_SIMPLE = 'https://assets.xrkmm.cn/u/4000002499670412/1b3b6895-9a51-4e4c-a4fd-b3878dbe987e.png'
const PRESENTED_IMAGE_HAPPY = 'https://assets.xrkmm.cn/u/4000002499670412/d0c4ce46-2459-4e8e-a6f4-861e842aba43.png'
const PRESENTED_IMAGE_ERROR = 'https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png'
const PRESENTED_IMAGE_MONEY = 'https://assets.xrkmm.cn/u/4000002499670412/42799481-a708-4157-82bc-c32afa46df7f.png'

export interface EmptyProps {
  src?: string
  size?: 'large' | 'middle' | 'small'
  text?: string
  wrapperTop?: number | string
  wrapperBottom?: number | string
}

type EmptyComponents = React.FC<EmptyProps> & {
  PRESENTED_IMAGE_DEFAULT: string
  PRESENTED_IMAGE_SIMPLE: string
  PRESENTED_IMAGE_HAPPY: string
  PRESENTED_IMAGE_ERROR: string
  PRESENTED_IMAGE_MONEY: string
}

const Empty: EmptyComponents = ({
  src = PRESENTED_IMAGE_DEFAULT,
  size = 'middle',
  text,
  wrapperTop,
  wrapperBottom,
  children
}) => {
  const _class = classNames('xrk-empty', `xrk-empty-${size}`)

  const _style: React.CSSProperties = {
    paddingTop: typeof wrapperTop === 'string' ? wrapperTop : transformRem(wrapperTop),
    paddingBottom: typeof wrapperBottom === 'string' ? wrapperBottom : transformRem(wrapperBottom)
  }

  return (
    <View className={_class} style={_style} data-testid="empty" >
      <View>
        <Image src={src} className="xrk-empty-img" mode="aspectFit" />
      </View>
      {Boolean(text) && (
        <Text className="xrk-empty-text">{text}</Text>
      )}
      {Boolean(children) && (
        <View className="xrk-empty-children">{children}</View>
      )}
    </View>
  )
}

Empty.PRESENTED_IMAGE_DEFAULT = PRESENTED_IMAGE_DEFAULT
Empty.PRESENTED_IMAGE_SIMPLE = PRESENTED_IMAGE_SIMPLE
Empty.PRESENTED_IMAGE_HAPPY = PRESENTED_IMAGE_HAPPY
Empty.PRESENTED_IMAGE_ERROR = PRESENTED_IMAGE_ERROR
Empty.PRESENTED_IMAGE_MONEY = PRESENTED_IMAGE_MONEY

export default Empty
