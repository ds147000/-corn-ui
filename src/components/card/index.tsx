/*
 * @Description:
 * @Author: zhoulong.yang
 * @Date: 2021-06-07 12:03:34
 * @LastEditors: zhoulong.yang
 * @LastEditTime: 2021-06-07 15:43:55
 */
import React from 'react'
import { View, Text } from '@tarojs/components'

const Card: React.FC = () => {
  return (
    <View>
      <Text>卡片</Text>
    </View>
  )
}

export default Card
