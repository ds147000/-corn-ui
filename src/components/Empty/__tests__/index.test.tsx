import React from 'react'
import { View } from '@tarojs/components'
import { render } from '@testing-library/react'
import Empty from '../index'

describe('Empty', () => {

  test('basis', () => {
    const screen = render(<Empty />)
    expect(screen).toMatchSnapshot()
  })

  test('text', () => {
    const screen = render(<Empty text="暂停服务..." />)
    screen.getByText('暂停服务...')
    expect(screen).toMatchSnapshot()
  })

  test('children', () => {
    const screen = render(
      <Empty>
        <View>我是子对象</View>
      </Empty>
    )
    screen.getByText('我是子对象')
    expect(screen).toMatchSnapshot()
  })

  test('img', () => {
    const screen = render(<Empty src="https://t7.baidu.com/it/u=1831997705,836992814&fm=193&f=GIF" />)
    expect(screen).toMatchSnapshot()
  })

  test('wrapperTop', () => {
    const screen = render(<Empty wrapperTop={100} />)
    expect(screen.getByTestId('empty').style.paddingTop).toBe('1.33rem')
    expect(screen).toMatchSnapshot()
  })

  test('wrapperTop to string', () => {
    const screen = render(<Empty wrapperTop="100px" />)
    expect(screen.getByTestId('empty').style.paddingTop).toBe('100px')
    expect(screen).toMatchSnapshot()
  })

  test('wrapperBottom', () => {
    const screen = render(<Empty wrapperBottom={100} />)
    expect(screen.getByTestId('empty').style.paddingBottom).toBe('1.33rem')
    expect(screen).toMatchSnapshot()
  })

  test('wrapperBottom ot string', () => {
    const screen = render(<Empty wrapperBottom="100px" />)
    expect(screen.getByTestId('empty').style.paddingBottom).toBe('100px')
    expect(screen).toMatchSnapshot()
  })
})
