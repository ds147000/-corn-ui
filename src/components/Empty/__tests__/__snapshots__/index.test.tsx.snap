// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty basis 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-empty xrk-empty-middle"
        data-testid="empty"
      >
        <div
          class=""
        >
          <div
            class="taro-img xrk-empty-img"
          >
            <img
              class="taro-img__mode-aspectfit"
              src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-empty xrk-empty-middle"
      data-testid="empty"
    >
      <div
        class=""
      >
        <div
          class="taro-img xrk-empty-img"
        >
          <img
            class="taro-img__mode-aspectfit"
            src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Empty children 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-empty xrk-empty-middle"
        data-testid="empty"
      >
        <div
          class=""
        >
          <div
            class="taro-img xrk-empty-img"
          >
            <img
              class="taro-img__mode-aspectfit"
              src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
            />
          </div>
        </div>
        <div
          class="xrk-empty-children"
        >
          <div
            class=""
          >
            我是子对象
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-empty xrk-empty-middle"
      data-testid="empty"
    >
      <div
        class=""
      >
        <div
          class="taro-img xrk-empty-img"
        >
          <img
            class="taro-img__mode-aspectfit"
            src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
          />
        </div>
      </div>
      <div
        class="xrk-empty-children"
      >
        <div
          class=""
        >
          我是子对象
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Empty img 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-empty xrk-empty-middle"
        data-testid="empty"
      >
        <div
          class=""
        >
          <div
            class="taro-img xrk-empty-img"
          >
            <img
              class="taro-img__mode-aspectfit"
              src="https://t7.baidu.com/it/u=1831997705,836992814&fm=193&f=GIF"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-empty xrk-empty-middle"
      data-testid="empty"
    >
      <div
        class=""
      >
        <div
          class="taro-img xrk-empty-img"
        >
          <img
            class="taro-img__mode-aspectfit"
            src="https://t7.baidu.com/it/u=1831997705,836992814&fm=193&f=GIF"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Empty text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-empty xrk-empty-middle"
        data-testid="empty"
      >
        <div
          class=""
        >
          <div
            class="taro-img xrk-empty-img"
          >
            <img
              class="taro-img__mode-aspectfit"
              src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
            />
          </div>
        </div>
        <span
          class="taro-text xrk-empty-text"
        >
          暂停服务...
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-empty xrk-empty-middle"
      data-testid="empty"
    >
      <div
        class=""
      >
        <div
          class="taro-img xrk-empty-img"
        >
          <img
            class="taro-img__mode-aspectfit"
            src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
          />
        </div>
      </div>
      <span
        class="taro-text xrk-empty-text"
      >
        暂停服务...
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Empty wrapperBottom 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-empty xrk-empty-middle"
        data-testid="empty"
        style="padding-bottom: 1.33rem;"
      >
        <div
          class=""
        >
          <div
            class="taro-img xrk-empty-img"
          >
            <img
              class="taro-img__mode-aspectfit"
              src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-empty xrk-empty-middle"
      data-testid="empty"
      style="padding-bottom: 1.33rem;"
    >
      <div
        class=""
      >
        <div
          class="taro-img xrk-empty-img"
        >
          <img
            class="taro-img__mode-aspectfit"
            src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Empty wrapperBottom ot string 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-empty xrk-empty-middle"
        data-testid="empty"
        style="padding-bottom: 100px;"
      >
        <div
          class=""
        >
          <div
            class="taro-img xrk-empty-img"
          >
            <img
              class="taro-img__mode-aspectfit"
              src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-empty xrk-empty-middle"
      data-testid="empty"
      style="padding-bottom: 100px;"
    >
      <div
        class=""
      >
        <div
          class="taro-img xrk-empty-img"
        >
          <img
            class="taro-img__mode-aspectfit"
            src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Empty wrapperTop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-empty xrk-empty-middle"
        data-testid="empty"
        style="padding-top: 1.33rem;"
      >
        <div
          class=""
        >
          <div
            class="taro-img xrk-empty-img"
          >
            <img
              class="taro-img__mode-aspectfit"
              src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-empty xrk-empty-middle"
      data-testid="empty"
      style="padding-top: 1.33rem;"
    >
      <div
        class=""
      >
        <div
          class="taro-img xrk-empty-img"
        >
          <img
            class="taro-img__mode-aspectfit"
            src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Empty wrapperTop to string 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="xrk-empty xrk-empty-middle"
        data-testid="empty"
        style="padding-top: 100px;"
      >
        <div
          class=""
        >
          <div
            class="taro-img xrk-empty-img"
          >
            <img
              class="taro-img__mode-aspectfit"
              src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="xrk-empty xrk-empty-middle"
      data-testid="empty"
      style="padding-top: 100px;"
    >
      <div
        class=""
      >
        <div
          class="taro-img xrk-empty-img"
        >
          <img
            class="taro-img__mode-aspectfit"
            src="https://assets.xrkmm.cn/u/4000002499670412/7d17be13-fb93-4b73-9027-b50ea6e4c236.png"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
